src/Propel/Runtime/Connection/ConnectionFactory.php

```diff
@@ -55,6 +55,11 @@ public static function create(array $configuration = [], AdapterInterface $adapt
                     }
                     $value = constant($value);
                 }
+                
+                if ($option == "ATTR_EMULATE_PREPARES" || $option == "ATTR_TIMEOUT") {
```

Comment:
ok, then maybe the check is better placed in Connection::setAttribute() which has the DSN and knows whether its MSSQL or not